hcl2template/types.build.go

```diff
@@ -150,6 +135,21 @@ func (p *Parser) decodeBuildConfig(block *hcl.Block, cfg *PackerConfig) (*BuildB
 	}
 	for _, block := range content.Blocks {
 		switch block.Type {
+		case buildHCPPackerRegistryLabel:
+			hcpPackerRegistry, moreDiags := p.decodeHCPRegistry(block)
+			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
+			if build.HCPPackerRegistry != nil {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  fmt.Sprintf("Only one " + buildHCPPackerRegistryLabel + " is allowed"),
+					Subject:  block.DefRange.Ptr(),
+				})
+				continue
+			}
```

Comment:
I'd put that check before calling for `decodeHCPRegistry`, so that we fail quick and don't potentially run other things.

Reply:
It could also be a good exercise to allow pushing to more than one thing, but probably out of our scope.